Er図作成

tga-business opened the pull request:
ER図、READMEに必要なデータベースの記述を書きました。

tga-business committed:
basic認証実装（途中）

tga-business committed:
basic_auth導入

tga-business committed:
ER図、README記述

tga-business committed:
Update README.md

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,58 @@
+##user
+|Column     |Typ    |Options |
+|-----------|-------|--------|
+|email      |string |NOT NULL|
```
同じメールアドレスでは登録できないことを明示するため、オプションに`unique: true`を記述しましょう！

また、`NOT NULL`ではなく`null: false`という記述を使用しましょう！(他のカラムについても同様です！)

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,58 @@
+|Column     |Typ    |Options |
+|-----------|-------|--------|
+|email      |string |NOT NULL|
+|password   |string |NOT NULL|
```
`|password   |string |NOT NULL|`
こちらのカラムはdeviseのgemを使用しての実装になるため、deviseが予め用意している
encrypted_passwordというカラムを代わりに用意してあげましょう。

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,58 @@
+|last_name  |string |NOT NULL|
+|first_kana |string |NOT NULL|
+|last_kana  |string |NOT NULL|
+birth_year  |      |         |
```
誕生日はdate型を使用することで、年月日を1つのカラムで保存することができます！

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,58 @@
+##product
+  |Column   |Type     |Options |
+  |---------|---------|--------|
+  |title    |string   |NOT NULL|
```
見本アプリを参考に、商品出品時に必要なカラムを追加しましょう！
(商品説明、商品の状態など)

tga-business replied:
それぞれをactive_hashを使用したモデルを使用することで分けてあります
状態がstatus
都道府県情報をfrom
発送日時の選択をesd
発送料の負担をburden
になります
これらをモデルで分け、選択されたものをテーブルに保存させるためにそれぞれカラムを作成する。
ということでしょうか？

shodai-ito replied:
返信ありがとうございます！
はい、その認識で問題ありません。
もしactive_hashの扱いで疑問等ありましたら、下記のカリキュラムを参考にしてください！
https://master.tech-camp.in/curriculums/5945

chobi555 replied:
ActivehashのカラムをDBのカラムとして定義しましょう。
わからないことがございましたら、コネクトにて質問していただくようお願いいたします。

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,58 @@
 # README
```
購入履歴を管理するテーブル、購入時に入力する住所を保存するテーブルを設計しましょう！

購入履歴を管理するテーブルに必要なカラムはuserとproductの外部キーカラムです。
userカラムは購入者を、productカラムは購入された商品と紐づきます。

住所テーブルに必要なカラムは、購入時に登録する情報です。
こちらは見本アプリを参考にしてみましょう！

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,58 @@
+
+##user
+|Column     |Typ    |Options |
+|-----------|-------|--------|
```
ニックネームのカラムも追加しましょう！

s-aka-moto reviewed (commented):
レビュー依頼ありがとうございます！
コメントいたしましたので、ご確認をお願いいたします！

また、次回レビュー依頼のご提出時にはER図の添付をお願いいたします！
What,Whyを記述する欄にgyazoのURLを記載いただくようお願いいたします。

tga-business committed:
readme編集

shodai-ito reviewed (commented):
レビュー依頼ありがとうございます！

また、レビュー内容に対する返信ありがとうございます！
コメントさせていただきましたので、ご確認をお願いします。

また、いただいたレビュー依頼ですが、返信いただいた箇所以外の修正点の差分があがっておりません。
コミットとプッシュのし忘れを確認して修正点の差分を出して再度ご提出をお願いいたします。

tga-business committed:
ER図更新

chobi555 commented on README.md:
```diff
@@ -1,24 +1,78 @@
+|last_name            |string  |null:  false|
+|first_kana           |string  |null:  false|
+|last_kana            |string  |null:  false|
+|birth                |integer |dat         |
```
`|birth                |integer |dat         |`
date型に変更しておきましょう。

chobi555 commented on README.md:
```diff
@@ -1,24 +1,78 @@
+credit
+|Column          |Type   |Options |
+|----------------|-------|--------|
+|deadline_year   |       |        |
+|deadline_month  |       |        |
+|security_code   |       |        |
```
クレジットカード情報はpayjpを用いて保存するのでカラムを設ける必要がありません。
消しておきましょう。

chobi555 commented on README.md:
```diff
@@ -1,24 +1,78 @@
+|deadline_year   |       |        |
+|deadline_month  |       |        |
+|security_code   |       |        |
+|postal_code     |       |        |
+|from_city       |       |        |
+|from_address    |       |        |
+|phone_number    |       |        |
```
```
|postal_code     |       |        |
|from_city       |       |        |
|from_address    |       |        |
|phone_number    |       |        |
```
こちらのカラムに型とOPTIONSを付与しておきましょう。

chobi555 commented on README.md:
```diff
@@ -1,24 +1,78 @@
+##Purchase history
+|Column               |Type    |Options |
+|---------------------|--------|------------|
+|product_id           |string  |unique: true|
+|user_id              |        |
```
```
|product_id           |string  |unique: true|
|user_id              |        |
```
こちらは外部キーですのでstiring型ではなく、別の型に変更しておきましょう。
またOPTIONSへの記述もお願いいたします。

chobi555 replied:
下記URLのコメント内容を実装しましょう。

https://github.com/tga-business/furima-34748/pull/2/files#r645931874

chobi555 commented on README.md:
```diff
@@ -1,24 +1,78 @@
+|birth                |integer |dat         |
+
+
+credit
```
こちらのテーブルにActivehashカラムが存在するかと思いますの、追記しておきましょう。

chobi555 reviewed (commented):
レビュー依頼ありがとうございます！
コメント致しました。
確認後、修正をお願いいたします。

tga-business committed:
er図編集２

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,79 @@
+##user
+|Column               |Type    |Options |
+|---------------------|--------|------------|
+|nickname             |string  |unique: true|
+|email                |string  |unique: true|
```
nicknameもemailも空を通してしまっては困るため、null: falseも記述しておきましょう！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,79 @@
+|first_kana           |string  |null:  false|
+|last_kana            |string  |null:  false|
+|birth                |integer |date        |
+
```
userテーブルのアソシエーションを記述しましょう！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,79 @@
+credit
+|Column          |Type   |Options |
+|----------------|-------|--------|
+|card_number     |integer       |null:  false        |
+|deadline_year   |integer       |null:  false        |
+|deadline_month  |integer       |null:  false        |
+|security_code   |integer  |下四桁、もしくは３桁のメソッド        |
```
カード情報を直接DBに保存するのは危険です。
こちらのカラムは不要ですので削除しておきましょう！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,79 @@
+|deadline_year   |integer       |null:  false        |
+|deadline_month  |integer       |null:  false        |
+|security_code   |integer  |下四桁、もしくは３桁のメソッド        |
+|postal_code     |integer       |null:  false|
```
郵便番号はstring型にしましょう！
integer型ですとハイフンを認識できません！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,79 @@
+|postal_code     |integer       |null:  false|
+|from_city       |string       |null:  false|
+|from_address    |integer       |        |
+|phone_number    |integer       |null:  false|
```
電話番号もstring型にしましょう！
integer型ですと、0から始まる数字を認識できません！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,79 @@
+|from_city       |string       |null:  false|
+|from_address    |integer       |        |
+|phone_number    |integer       |null:  false|
+has_many :from
```
住所の入力で求められるのは、
・郵便番号
・都道府県
・市町村
・番地
・建物名（任意）
・電話番号
です。
カラムが不足しているかと思いますので、必要なものを追加しましょう！

chobi555 replied:
まだ不足しているカラムが存在するかと思いますので、見本アプリを参考に実装しましょう！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,79 @@
+|from_city       |string       |null:  false|
+|from_address    |integer       |        |
+|phone_number    |integer       |null:  false|
+has_many :from
```
formというテーブルは存在していないかと存じます。
purchase_historyとのアソシエーション に変更しておきましょう！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,79 @@
+##Purchase history
+|Column               |Type    |Options |
+|---------------------|--------|------------|
+|product_id           |string  |unique: true|
+|user_id              |        |
```
こちら外部キーですので、references型を使用しましょう！
その際、外部キー制約も忘れずに記述しておきましょう！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,79 @@
+  |---------|---------|--------|
+  |title    |string   |NOT NULL|
+  |category |text     |NOT NULL|
+  |price    |integer  |NOT NULL|
```
active_hashのカラムが実装途中かと存じます。
productテーブルに状態、都道府県情報、発送日時、発送料の負担のカラムを追加しましょう！

chobi555 replied:
下記URLのコメント内容を実装しましょう！
わからないことがございましたら、気軽にコネクトにて質問していただくようお願いいたします。

https://github.com/tga-business/furima-34748/pull/2/files#r645937739

shodai-ito reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

その後、再度レビュー依頼フォームのご提出をよろしくお願いいたします。

tga-business committed:
er図編集３

chobi555 commented on README.md:
```diff
@@ -1,24 +1,78 @@
+##user
+|Column               |Type    |Options |
+|---------------------|--------|------------|
+|nickname             |string  |unique: true|
+|email                |string  |unique: true|
```
```
|nickname             |string  |unique: true|
|email                |string  |unique: true|
```
OPTIONSにnull:  falseを追記しておきましょう。

chobi555 commented on README.md:
```diff
@@ -1,24 +1,78 @@
+|last_name            |string  |null:  false|
+|first_kana           |string  |null:  false|
+|last_kana            |string  |null:  false|
+|birth                |integer |date        |
```
`|birth                |integer |date        |`
Typeの部分をdateに変更しておきましょう。
また、OPTIONSにはnull: falseを記述しておきましょう。

chobi555 commented on README.md:
```diff
@@ -1,24 +1,78 @@
+credit
+|Column          |Type   |Options |
+|----------------|-------|--------|
+|deadline_year   |integer       |null:  false        |
+|deadline_month  |integer       |null:  false        |
+|security_code   |integer  |下四桁、もしくは３桁のメソッド        |
```
クレジットカード情報はpayjpを用いて保存するので、カラムを設ける必要はございません。
消しておきましょう。

chobi555 reviewed (commented):
レビュー依頼ありがとうございます！
コメント致しました。
確認後、修正をお願いいたします。

tga-business committed:
ER図編集

テーブル設計における基本的なテーブル数はこれで問題ないでしょうか？

chobi555 commented on README.md:
```diff
@@ -1,24 +1,76 @@
+|first_kana           |string  |null: false|
+|last_kana            |string  |null: false|
+|birth                |date    |null: false|
+|product_id           |date    |null: false|
```
`|product_id           |date    |null: false|`
こちらは不要かと思いますので、消しておきましょう。

chobi555 commented on README.md:
```diff
@@ -1,24 +1,76 @@
+  |title                 |string   |null: false|
+  |category              |text     |null: false|
+  |price                 |integer  |null: false|
+  |status_id             |integer  |null: false|
+  |delivery_charge_id    |integer  |null: false|
+  |from_id               |integer  |null: false|
+  |burden_id             |integer  |null: false|
```
Activehashカラムがあと一つ存在するかと思いますので、追記しておきましょう。

chobi555 commented on README.md:
```diff
@@ -1,24 +1,76 @@
-application up and running.
 
-Things you may want to cover:
+##product
```
productテーブルにUsersテーブルの外部キーを定義しましょう。

chobi555 commented on README.md:
```diff
@@ -1,24 +1,76 @@
+|last_kana            |string  |null: false|
+|birth                |date    |null: false|
+|product_id           |date    |null: false|
+
```
Usersテーブルのアソシエーションを記述しましょう。

chobi555 commented on README.md:
```diff
@@ -1,24 +1,76 @@
+##Purchase history(購入履歴)
+|Column               |Type    |Options |
+|---------------------|--------|------------|
+|product_id           |integer  |
+|user_id              |integer  |
```
OPTIONSに外部キー制約を追記しておきましょう。

chobi555 commented on README.md:
```diff
@@ -1,24 +1,76 @@
+|----------------|-------|--------|
+|postal_code     |string       |null:  false|
+|from_city       |string       |null:  false|
+|from_address    |integer      |null:  false|
```
`|from_address    |integer      |null:  false|`
こちらはActivehashのカラムでしょうか？
Activehashのカラムであれば、カラム名の末尾に`_id`を付与しましょう。
また、productテーブルに同じ内容のカラムが存在するかと思いますので、
カラム名をどちらかに統一させておきましょう。

chobi555 commented on README.md:
```diff
@@ -1,24 +1,76 @@
+|product_id           |date    |null: false|
+
+
+##住所
```
こちらのテーブルにPurchase historyの外部キーを定義しましょう。

chobi555 reviewed (commented):
レビュー依頼ありがとうございます！
コメント致しました。
確認後、修正をお願いいたします。

tga-business committed:
ER

tga-business committed:
外部キー設定、購入履歴、住所テーブルのアソシエーション設定

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,83 @@
+  |from_id               |integer  |null: false      |
+  |burden_id             |integer  |null: false      |
+  |esd_id                |integer  |null: false      |
+  |user_id               |integer  |foreign_key: true|
```
外部キーカラムは型をreferences型に修正し、カラム名の「_id」を削除しましょう！
references型の場合、自動的に_idが付与されるためです。

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,83 @@
+  |Column                |Type     |Options          |
+  |----------------------|---------|-----------------|
+  |title                 |string   |null: false      |
+  |category              |text     |null: false      |
```
商品説明カラムがないかと思われます。
また不要なカラムはないか、見本アプリを参考に再度考えてみましょう！

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,83 @@
+|first_kana           |string  |null: false              |
+|last_kana            |string  |null: false              |
+|birth                |date    |null: false              |
+has_one :street_address
```
ユーザーテーブルは住所テーブルと紐づかないためアソシエーションを削除しましょう！
その代わり購入管理テーブルとのアソシエーションをお願いいたします。

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,83 @@
+|----------------|-------------|------------|
+|postal_code     |string       |null:  false|
+|from_city       |string       |null:  false|
+|from_address    |integer      |null:  false|
```
https://github.com/tga-business/furima-34748/pull/2/files#r645951300
こちらが未修正のため修正をお願いいたします。

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,83 @@
+|from_address    |integer      |null:  false|
+|building        |string       |            |
+|phone_number    |string       |null:  false|
+|from_id         |references   |null:  false|
```
`|from_id         |references   |null:  false|
`
fromテーブルは存在しないため、こちらのカラムは削除しましょう。
またその代わりとして購入管理テーブルの外部キーカラムを記述しましょう！

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,83 @@
+|---------------------|------------|-----------------|
+|product_id           |references  |foreign_key: true|
+|user_id              |references  |foreign_key: true|
+has_many :users
+has_many :products
+has_one  :street_address
```
```
has_many :users
has_many :products
```
上記のアソシエーションですと「一つの購入履歴に複数のユーザーと複数の商品」が紐づいていることになります。
適切な関係になるよう修正をお願いいたします！

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,83 @@
+##purchase history(購入履歴)
+|Column               |Type        |Options          |
+|---------------------|------------|-----------------|
+|product_id           |references  |foreign_key: true|
+|user_id              |references  |foreign_key: true|
```
カラム名の「_id」を削除しましょう！
reference型を使用するとカラムに「_id」が自動で付与されるためです。

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,83 @@
-application up and running.
 
-Things you may want to cover:
+##product
```
各テーブル名は複数形で記述をし、スペースが入る場合はアンダーバーで繋ぎましょう！

Saka-toku reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
コメントいたしましたので、ご確認お願いいたします！

tga-business committed:
商品説明テーブル追加、fromモデルをprefecutureに変更、アソシエーション、カラム修正

・active_hashをintegerタイプで使用する場合、_idを記述する

・references型は_idが自動で付与されるので_idをカラムにつけなくても良い

購入履歴にユーザー情報を獲得してあるので住所テーブルとユーザーテーブルは紐付けない

tga-business committed:
テーブルを複数形に変更

div-suzuki commented on README.md:
```diff
@@ -1,24 +1,89 @@
+belongs_to :product
 
-* System dependencies
+# status
+active_hash使用
+商品の状態についての選択肢を記述予定
+belongs_to :product
 
-* Configuration
+# prefecture
+active_hash使用
+都道府県記述
 
-* Database creation
 
-* Database initialization
+# esd
+active_hash使用
+発送日時の選択肢について記述
 
-* How to run the test suite
 
-* Services (job queues, cache servers, search engines, etc.)
+# burden
+active_hash使用
+発送料の負担について記述
```
アクティブハッシュに関しては上記と同様の理由で、
こちらの記述は消しておきましょう。

div-suzuki commented on README.md:
```diff
@@ -1,24 +1,89 @@
+  belongs_to :user
+  belongs_to :purchase_history
+  has_many   :comments
+  has_many   :status
+  has_many   :delivery_charge
+  has_many   :prefecture
+  has_many   :burden
+  has_many   :esd
```
後ほど詳しく学習をされると思いますが、アクティブハッシュではテーブルを必要としません。
よって、アソシエーションの記述は不要になります。

div-suzuki commented on README.md:
```diff
@@ -1,24 +1,89 @@
+  |Column                |Type        |Options          |
+  |----------------------|------------|-----------------|
+  |title                 |string      |null: false      |
+  |description           |string      |null: false      |
```
``
  |description           |string      |null: false      |
``
商品説明は255字を超える可能性があるため、text型で実装しましょう！

div-suzuki commented on README.md:
```diff
@@ -1,24 +1,89 @@
+|first_kana           |string  |null: false              |
+|last_kana            |string  |null: false              |
+|birth                |date    |null: false              |
+belongs_to :purchase_history
```
``
belongs_to :purchase_history
``
こちらユーザーは複数の購入が可能だと思うので、has_manyを使用しましょう。

div-suzuki commented on README.md:
```diff
@@ -1,24 +1,89 @@
+|----------------|-------------|------------|
+|postal_code     |string       |null:  false|
+|from_city       |string       |null:  false|
+|from_address    |integer      |null:  false|
```
https://github.com/tga-business/furima-34748/pull/2/files#r645951300

こちらのご確認もよろしくお願いします！

div-suzuki commented on README.md:
```diff
@@ -1,24 +1,89 @@
+|building        |string       |            |
+|phone_number    |string       |null:  false|
+|prefecture_id   |references   |null:  false|
+ has_many :prefecture
```
``
 has_many :prefecture
``
アクティブハッシュに関しては上記と同様の理由で、
こちらの記述は消しておきましょう。

div-suzuki commented on README.md:
```diff
@@ -1,24 +1,89 @@
+|---------------------|------------|-----------------|
+|product              |references  |foreign_key: true|
+|user                 |references  |foreign_key: true|
+has_one :users
+has_one :products
```
今回は1対多の関係のため、こちらにはbelongs_toを使用しましょう。

div-suzuki reviewed (commented):
レビュー依頼ありがとうございます。
修正点確認させていただきました！
追加でコメントいたしましたので、ご確認お願いいたします。

tga-business commented:
from_addressカラムは番地のカラムなのでactive_hashではありません。
都道府県のactive_hashをfromからprefectureに変更してありますので解決しているかとfrom_addressの問題は解決しているかと思われます。

カリキュラムを見たところ、active_hashモデル同士のアソシエーションは必要ということでテーブルを作成する際のモデルの補足ということで残してあります。

tga-business committed:
アソシエーション編集、前回のレビューに対しgithubにコメント送信

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,89 @@
+  |esd_id                |integer     |null: false      |
+  |user                  |references  |foreign_key: true|
+  belongs_to :user
+  has_many :purchase_history
```
```ruby
has_many :purchase_history
```
1つの商品は1回のみ購入できますので、has_manyではなく、`has_one`で記述しましょう。

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,89 @@
+  has_many :purchase_history
+  has_many   :comments
+  has_many   :status
+  has_many   :delivery_charge
+  has_many   :prefecture
+  has_many   :burden
+  has_many   :esd
```
アソシエーションの記述を残すのであれば、
has_manyで記述していますので、複数形で記述しましょう。

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,89 @@
+  |----------------------|------------|-----------------|
+  |title                 |string      |null: false      |
+  |description           |text        |null: false      |
+  |category              |integer     |null: false      |
+  |price                 |integer     |null: false      |
+  |status_id             |integer     |null: false      |
+  |delivery_charge_id    |integer     |null: false      |
+  |prefecture_id         |integer     |null: false      |
+  |burden_id             |integer     |null: false      |
+  |esd_id                |integer     |null: false      |
```
```ruby
|category |integer |null: false |
```
こちら、active_hashカラムであれば
```
カラム名(単数形)の末尾に_idを追加
```
しましょう。


```ruby
|delivery_charge_id |integer |null: false |
```
またこちらのカラムは必要でしょうか？
active_hashのカラムは以下の5つになります。
```
商品の状態
配送料の負担
発送元の地域
発送までの日数
カテゴリー
```

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,89 @@
+|first_kana           |string  |null: false              |
+|last_kana            |string  |null: false              |
+|birth                |date    |null: false              |
+has_many :purchase_history
```
```ruby
has_many :purchase_history
```
has_manyでアソシエーションを記述する場合は、複数形で記述しましょう。
```ruby
has_many: 複数形
```

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,89 @@
+|----------------|-------------|------------|
+|postal_code     |string       |null:  false|
+|from_city       |string       |null:  false|
+|from_address    |integer      |null:  false|
```
```ruby
|from_address |integer |null: false|
```
番地のカラムですが、数字以外にハイフン等の入力も考えられますので、
`string型`にしておきましょう。

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,89 @@
+|from_address    |integer      |null:  false|
+|building        |string       |            |
+|phone_number    |string       |null:  false|
+|prefecture_id   |references   |null:  false|
```
```ruby
|prefecture_id |references |null: false|
```
active_hashのカラムは、references型ではなく`integer型`に修正しましょう。

Hide-Morishita replied:
street_addressテーブルはpurchase_historyテーブルとのアソシエーションを結んでいますので、purchase_historyテーブルのidを参照する**外部キー**を記述しましょう。

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,89 @@
+|building        |string       |            |
+|phone_number    |string       |null:  false|
+|prefecture_id   |references   |null:  false|
+ has_many :prefecture
```
```ruby
has_many :prefecture
```
こちら記述を残すのであれば、以下のようにしましょう。
has_manyでアソシエーションを記述する場合は、複数形で記述しましょう。
```ruby
has_many: 複数形
```

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,89 @@
+|prefecture_id   |references   |null:  false|
+ has_many :prefecture
+ belongs_to :purchase_history
+ has_many :prefecture
```
```ruby
has_many :prefecture
```
76行目と同様の記述になっています。
不要になりますので削除しましょう。

Hide-Morishita commented on README.md:
```diff
@@ -1,24 +1,89 @@
+|---------------------|------------|-----------------|
+|product              |references  |foreign_key: true|
+|user                 |references  |foreign_key: true|
+ belongs_to :users
+ belongs_to :products
```
```ruby
belongs_to :users
 belongs_to :products
```
belongs_toで記述する場合は、単数形で記述しましょう。

Hide-Morishita reviewed (commented):
お待たせいたしました、レビュー依頼ありがとうございます。
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いいたします。
コードの修正後、コミット・プッシュを行った上で再度レビュー依頼をお送りいただくようお願いいたします。

次回レビュー依頼の際に、READMEの内容を反映させた、最新のER図の提出をお願いいたします。

tga-business committed:
アソシエーション変更

saki125 commented on README.md:
```diff
@@ -1,24 +1,88 @@
+  |----------------------|------------|-----------------|
+  |title                 |string      |null: false      |
+  |description           |text        |null: false      |
+  |category              |integer     |null: false      |
```
```ruby
|category              |integer     |null: false      |
```
こちらのカラムもアクティブハッシュでの実装かと存じます。他のカラムと同様カラム名に`_id`をつけましょう。

saki125 commented on README.md:
```diff
@@ -1,24 +1,88 @@
+|----------------|-------------|------------|
+|postal_code     |string       |null:  false|
+|from_city       |string       |null:  false|
+|from_address    |integer      |null:  false|
```
```ruby
|from_address |integer |null: false|
```
番地のカラムですが、数字以外にハイフン等の入力も考えられますので、
`string型`にしておきましょう。

saki125 commented on README.md:
```diff
@@ -1,24 +1,88 @@
+|phone_number    |string       |null:  false|
+|prefecture_id   |integer      |null:  false|
+|purchase_history              |references  |foreign_key: true|
+ has_many :prefectures
+ belongs_to :purchase_history
+ has_many :prefectures
```
```ruby
has_many :prefectures
```
こちらの記述が２回ございます。１つ削除しておきましょう。

saki125 reviewed (commented):
レビュー依頼ありがとうございます。
コメントいたしましたので、ご確認いただけますと幸いです。

修正後、再度レビュー依頼フォームのご提出をお願いいたします。
ER図も新しいものをご提出いただけますと幸いです。
お手数をお掛け致しますが、よろしくお願いいたします。

tga-business committed:
カラム記述編集

tga-business commented:
編集しプッシュしました

saki125 commented:
レビュー依頼ありがとうございます！
しっかりと修正されており、バッチリですね！
お手数ですが、実際のER図を拝見させていただきたいです。
ER図を修正し添付していただいた後、再度レビュー依頼フォームを送っていただいてもよろしいでしょうか？

お手数おかけしますが、よろしくお願いいたします。

tga-business commented:
https://i.gyazo.com/a662ed5b4aa06091ae5fd8847b411cb7.png

chobi555 commented on README.md:
```diff
@@ -1,24 +1,88 @@
+|phone_number    |string       |null:  false|
+|prefecture_id   |integer      |null:  false|
+|purchase_history              |references  |foreign_key: true|
+ has_many :prefectures
```
`has_many :prefectures`
こちらのアソシエーションは不要ですので、消しておきましょう！

chobi555 reviewed (commented):
レビュー依頼ありがとうございます！
コメント致しました。
確認後、修正をお願い致します！

あと少しです、頑張っていきましょう！

tga-business commented:
都道府県の選択は住所にも必要なので同じ用途として応用していますので削除する必要はないと思います

ken-1853 commented:
レビュー依頼とコメントありがとうございます。
LGTMとさせていただきます。実装お疲れさまでした！
念の為ご自身でもう一度挙動を確認いただき、問題なければマージしましょう！

```
都道府県の選択は住所にも必要なので同じ用途として応用していますので削除する必要はないと思います
```

こちらに関してですが、結論としては消さずにそのまま次の実装に進んでいただいて大丈夫です！

今回の指摘に至った経緯をご説明できればと思います。

まず前提として、readmeファイルというのはマイグレーションファイルに記述する内容を書いていただきます。
アクティブハッシュのアソシエーションに関しては、マイグレーションファイルには記述しないため、readmeファイルにも書かないことが多いです。
よって前回、```has_many :prefectures```のアソシエーションを削除しましょう！と言ったコメントをさせていただきました。

本来ですと、商品テーブルに記載いただいたアクティブハッシュのアソシエーションに関しても、同様のコメントをしなくてはいけないのですが、
それが漏れてしまったために、tga-business様の混乱を招くことになってしまいました。
大変申し訳ございませんでした。

ただ、モデルではアクティブハッシュのアソシエーションを組んでいただくことになるため、備忘録としてreadmeに記述を残しておくことはありです。

よって、今回はこちらでLGTMとさせていただきますので、tga-businessのご判断で、そのまま進めるか、記述をけして進めるかを選択いただければと思います。

なお修正後の再度レビュー提出は不要でございます。


マージ後、ローカルのmasterブランチに忘れずにプルしましょう。
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
引き続き頑張っていきましょう。応援しております！
